opm/parser/eclipse/EclipseState/IOConfig/IOConfig.cpp

```diff
@@ -39,29 +40,241 @@
+
+        /* If no BASIC mnemonic is found, either it is not present or we might
+         * have an old data set containing integer controls instead of mnemonics.
+         * BASIC integer switch is integer control nr 1, FREQUENCY is integer
+         * control nr 6.
+         */
+
+        /* mnemonics, but without basic and freq. Effectively ignored */
+        if( !ints ) return {};
+
+        if( items.size() >= 1 )
+            basic = std::stoul( items[ 0 ] );
+
+        // Peculiar special case in eclipse, - not documented
+        // This ignore of basic = 0 for the integer mnemonics case
+        // is done to make flow write restart file at the same intervals
+        // as eclipse for the Norne data set. There might be some rules
+        // we are missing here.
+        if( 0 == basic ) return {};
+
+        if( items.size() >= 6 ) // if frequency is set
+            freq = std::stoul( items[ 5 ] );
+
+        return restartConfig( step, basic, freq );
+    }
+
+    IOConfig::restartConfig IOConfig::rptsched( const DeckKeyword& keyword ) {
+        size_t restart = 0;
+        bool restart_found = false;
+
+        const auto& items = keyword.getStringData();
+        const auto ints = std::any_of( items.begin(), items.end(), is_int );
+        const auto strs = !std::all_of( items.begin(), items.end(), is_int );
+
+        if( ints && strs ) throw std::runtime_error(
+                "RPTSCHED does not support mixed mnemonics and integer list."
+            );
+
+        for( const auto& mnemonic : items  ) {
+            const auto restart_pos = mnemonic.find( "RESTART=" );
+            if( restart_pos != std::string::npos ) {
+                restart = std::stoul( mnemonic.substr( restart_pos + 8 ) );
+                restart_found = true;
+            }
+
+            const auto nothing_pos = mnemonic.find( "NOTHING" );
+            if( nothing_pos != std::string::npos ) {
+                restart = 0;
+                restart_found = true;
+            }
         }
+
+        if( restart_found ) return restartConfig( restart );
+
+
+        /* No RESTART or NOTHING found, but it is not an integer list */
+        if( strs ) return {};
+
+        /* We might have an old data set containing integer controls instead of
+         * mnemonics. Restart integer switch is integer control nr 7
+         */
+
+        if( items.size() < 7 ) return {};
+
+        return restartConfig( std::stoul( items[ 6 ] ) );
     }
 
+    DynamicState< IOConfig::restartConfig > IOConfig::rstconf(
+            const SCHEDULESection& schedule,
+            std::shared_ptr< const TimeMap > timemap ) {
+
+        size_t current_step = 1;
+        bool ignore_RPTSCHED_restart = false;
+        restartConfig unset;
+
+        DynamicState< IOConfig::restartConfig >
+            restart_config( timemap, restartConfig( 0, 0, 1 ) );
+
+        for( const auto& keyword : schedule ) {
+            const auto& name = keyword.name();
```

Comment:
The body in this loop is not very easy to read; I would have preferred if it was reorganized slightly so that two phases; i.e. check if we should update - and then the actual update were more clearly separated - know it is a bit of a blur.
```
    if (cond1)  continue;
    if (cond2)  continue;
    if (cond3)  continue;

    // OK - now we know that we should update the restart configuration; create the rs object and get going.
  
    rs = ....
```